### PR TITLE
Prepare for Cloud authentication support

### PR DESCRIPTION
**Pull request by daka1510**
#### Key changes
- extend public interface of `IBMRuntimeService` to prepare for Cloud and Legacy authentication support, i.e. add support for new `auth` and `locator` arguments to class constructor 
- updated existing test cases to use keyword arguments and explicitly pass in the `auth` type `legacy`

#### Minor refactorings
- extended Black to also format setup.py and added a `Code style: black` badge to the README
- sorted imports in `ibm_runtime_service` with [PyCharm](https://www.jetbrains.com/pycharm/guide/tips/optimize-imports/) (for later discussion: we might want to make use of an import sort utility that automaticall ensures we all use the same pattern)

### Details and comments
First step towards https://github.com/Qiskit/qiskit-ibm-runtime/issues/29.

### CODE REVIEW

File: qiskit_ibm_runtime/ibm_runtime_service.py [hunk 1]
```diff
@@ -127,13 +127,20 @@ class IBMRuntimeService:
...
+        **kwargs: Any,
     ) -> None:
         """IBMRuntimeService constructor
 
         Args:
-            token: IBM Quantum token.
-            url: URL for the IBM Quantum authentication server.
+            auth: Authentication type. `cloud` or `legacy`. If not specified, the saved default is used.
+                If there is no default value, and both accounts were saved on disk, the cloud type is
+                used.
```
Comment (rathishcholarajan) [inline qiskit_ibm_runtime/ibm_runtime_service.py:141]: Should we explain a bit here about what is cloud vs what is legacy?

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
@daka1510 Could you please resolve merge conflict?

**Comment by daka1510** [comment]
@rathishcholarajan resolved merge conflicts and added a comment. Can you take a look again?

**Comment by rathishcholarajan** [comment]
> @rathishcholarajan resolved merge conflicts and added a comment. Can you take a look again?

@daka1510 Thanks. I don’t see the comment though.

**Comment by daka1510** [comment]
Sorry, @rathishcholarajan - forgot to submit the review comment. Done.

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
@daka1510 Push tests are failing on your fork, could you please fix those issues? https://github.com/daka1510/qiskit-ibm-runtime/runs/4367753072?check_suite_focus=true#step:5:456

**Review by rathishcholarajan — APPROVED** [review]
@daka1510 Looks good!

**Pull request merged**